docs/design/restic.rst

```diff
@@ -0,0 +1,158 @@
+       # ReplicationSourceVolumeOptions
+
+       ### Restic-specific options
+       pruneIntervalDays:  # How often to prune the repository (*int)
+       repository:  # Secret name containing repository info (string)
+       # Retention policy for the backups
+       retain:
+         last:  # Keep the last n snapshots (*int)
+         hourly:  # Keep n hourly (*int)
+         daily:  # Keep n daily (*int)
+         weekly:  # Keep n weekly (*int)
+         monthly:  # Keep n monthly (*int)
+         yearly:  # Keep n yearly (*int)
+         within: # Keep all within this duration (e.g., "3w15h") (*string)
+
+The ``.spec.restic.repository`` Secret reference in the above structure refers
+to a Secret in the same Namespace of the following format. The Secret's "keys"
+correspond directly to the environment variables supported by Restic.
+
+.. code:: yaml
+
+   ---
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: resticRepo
+   type: Opaque
+   data:
+     # The repository url
+     RESTIC_REPOSITORY: s3:s3.amazonaws.com/bucket_name
+     # The repository encryption key
+     RESTIC_PASSWORD: XXXXX
+     # ENV vars specific to the back end
+     # https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html
+     AWS_ACCESS_KEY_ID: (access key)
+     AWS_SECRET_ACCESS_KEY: (secret key)
+
+Restore
+-------
+
+For now, with Scribe, the intention is to only support restoring the latest
+version of the backed-up data. For retrieving previous backups (that are still
+retained), Restic can be directly run against the repository, using the same
+information as in the Secret, above.
+
+Restore would be handled by the following ReplicationDestination:
+
+.. code:: yaml
+
+   ---
+   apiVersion: scribe.backube/v1alpha1
+   kind: ReplicationDestination
+   metadata:
+     name: dest-sample
+   spec:
+     trigger:
+       schedule: "30 * * * *"
+     restic:
+       ### Standard volume options
+       # ReplicationDestinationVolumeOptions
+
+       ### Restic-specific options
+       repository:  # Secret name containing repository info (string)
+
+There are comparatively few configuration options for Restore.
+
+Open issues
+===========
+
+The following items are open questions:
+
+- Should ReplicationDestination support scheduling or should it be based on a
+  single restore (i.e., it "syncs" once then never again)? This could also be
+  simulated by having an arbitrarily long schedule since the 1st sync is
+  immediate.
+
+- Are Restic operations fast enough to make this viable?
+
+  - The ``prune`` operation is documented as being rather slow
+
+  - How long does it take to scan the storage to determine what needs to be
+    backed up?
+
+- Restic uses locks on the repository. Does the lack of concurrency present a
+  problem for us? (Some can be done w/o locks... which ones?)
+
+- What is the right way to expose ``prune``?
+
+  - It is the method for freeing space in the repo, but may be too expensive to
+    run frequently
```

Comment:
`restic forget` is definitely faster than `restic prune` Explanation here https://github.com/restic/restic/issues/1335#issuecomment-334671989

We could set to prune twice a month. Also, single restore ReplicationDestinatio makes more sense given restic will not be ideally used for data replication.